test/e2e/framework/util.go

```diff
@@ -4945,17 +4946,49 @@ func (p *E2ETestNodePreparer) CleanupNodes() error {
 	return encounteredError
 }
 
-func CleanupGCEResources(loadBalancerName string) (err error) {
+// CleanupGCEResources cleans up GCE Service Type=LoadBalancer resources with
+// the given name. The name is usually the UUID of the Service prefixed with an
+// alpha-numeric character ('a') to work around cloudprovider rules.
+func CleanupGCEResources(loadBalancerName string) (retErr error) {
 	gceCloud, ok := TestContext.CloudConfig.Provider.(*gcecloud.GCECloud)
 	if !ok {
 		return fmt.Errorf("failed to convert CloudConfig.Provider to GCECloud: %#v", TestContext.CloudConfig.Provider)
 	}
-	gceCloud.DeleteFirewall(loadBalancerName)
-	gceCloud.DeleteForwardingRule(loadBalancerName)
-	gceCloud.DeleteGlobalStaticIP(loadBalancerName)
-	hc, _ := gceCloud.GetHttpHealthCheck(loadBalancerName)
-	gceCloud.DeleteTargetPool(loadBalancerName, hc)
-	return nil
+	if err := gceCloud.DeleteFirewall(loadBalancerName); err != nil &&
+		!IsGoogleAPIHTTPErrorCode(err, http.StatusNotFound) {
+		retErr = err
+	}
+	if err := gceCloud.DeleteForwardingRule(loadBalancerName); err != nil &&
+		!IsGoogleAPIHTTPErrorCode(err, http.StatusNotFound) {
+		retErr = fmt.Errorf("%v\n%v", retErr, err)
+
+	}
+	if err := gceCloud.DeleteGlobalStaticIP(loadBalancerName); err != nil &&
+		!IsGoogleAPIHTTPErrorCode(err, http.StatusNotFound) {
+		retErr = fmt.Errorf("%v\n%v", retErr, err)
+	}
+	// This function shells out to gcloud, so we can't compare for NotFound errors.
+	// TODO: Invoke cloudprovider method directly instead.
+	if err := DeleteGCEStaticIP(loadBalancerName); err != nil {
+		Logf("%v", err)
+	}
+	hc, getErr := gceCloud.GetHttpHealthCheck(loadBalancerName)
+	if getErr != nil && !IsGoogleAPIHTTPErrorCode(getErr, http.StatusNotFound) {
+		retErr = fmt.Errorf("%v\n%v", retErr, getErr)
+		return
```

Comment:
oh you mean why return without going thorugh to the end of this function? because we pass health checks as an argument to the next call `deleteTargetPools`